temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInternal.java

```diff
@@ -197,6 +210,41 @@ public static void registerListener(Object implementation) {
           .registerSignalHandlers(
               new WorkflowOutboundCallsInterceptor.RegisterSignalHandlersInput(requests));
     }
+    List<WorkflowOutboundCallsInterceptor.UpdateRegistrationRequest> updateRequests = new ArrayList<>();
+    for (POJOWorkflowMethodMetadata methodMetadata : workflowMetadata.getUpdateMethods()) {
+      Method method = methodMetadata.getWorkflowMethod();
+      UpdateMethod updateMethod = method.getAnnotation(UpdateMethod.class);
+
+      updateRequests.add(
+          new WorkflowOutboundCallsInterceptor.UpdateRegistrationRequest(
+              methodMetadata.getName(),
+              method.getParameterTypes(),
+              method.getGenericParameterTypes(),
+              (args) -> {
+                try {
+                  Validator<?> validator = updateMethod.validator().getDeclaredConstructor().newInstance();
+                  // TODO(quinn) cast implementation to <?> don't think this is possible
```

Comment:
@Spikhalskiy This is the part I wasn't sure was possible if the `Validator` had  a generic argument